cmd/prom-aggregation-gateway/main.go

```diff
@@ -206,15 +204,19 @@ func (a *aggate) parseAndMerge(r io.Reader) error {
 	a.familiesLock.Lock()
 	defer a.familiesLock.Unlock()
 	for name, family := range inFamilies {
-		// Sort labels in case source sends them inconsistently
-		for _, m := range family.Metric {
-			sort.Sort(byName(m.Label))
+		// Discard unsupported types
+		if *family.Type == dto.MetricType_SUMMARY {
+			continue
 		}
-
 		if err := validateFamily(family); err != nil {
 			return err
 		}
 
+		// Sort labels in case source sends them inconsistently
+		for _, m := range family.Metric {
+			sort.Sort(byName(m.Label))
+		}
+
```

Comment:
What's the reason to move this code?

Reply:
The idea was to keep the validation first before doing actual work